pkg/tnf/handlers/imagepullpolicy/imagepullpolicy_test.go

```diff
@@ -17,84 +17,124 @@
-func TestImagepullpolicy_GetIdentifier(t *testing.T) {
-	// Todo: Write test.
+func TestLogging_GetIdentifier(t *testing.T) {
+	test, handlers, jsonParseResult, err := createTest()
+
+	assert.Nil(t, err)
+	assert.True(t, jsonParseResult.Valid())
+	assert.NotNil(t, handlers)
+
+	assert.Equal(t, identifier.ImagePullPolicyIdentifier, (*test).GetIdentifier())
 }
 
-// Test_Imagepullpolicy_ReelFirst is the unit test for Imagepullpolicy_ReelFirst().
-func TestImagepullpolicy_ReelFirst(t *testing.T) {
-	// Todo: Write test.
+func TestLogging_ReelFirst(t *testing.T) {
+	_, handlers, jsonParseResult, err := createTest()
+
+	assert.Nil(t, err)
+	assert.True(t, jsonParseResult.Valid())
+	assert.NotNil(t, handlers)
+
+	assert.Equal(t, 1, len(handlers))
+	handler := handlers[0]
+	step := handler.ReelFirst()
+	expectedCommand := fmt.Sprintf("oc get pod %s -n %s -o json  | jq -r '.spec.containers[%d].imagePullPolicy'", testPodName, testPodNameSpace, testContainerNum)
+	assert.Equal(t, expectedCommand, step.Execute)
+	assert.Contains(t, step.Expect, expectedPassPattern, expectedFailPattern, expectedErrPattern)
+	assert.Equal(t, testTimeoutDuration, step.Timeout)
 }
 
-// Test_Imagepullpolicy_ReelEOF is the unit test for Imagepullpolicy_ReelEOF().
-func TestImagepullpolicy_ReelEOF(t *testing.T) {
-	handler := imagepullpolicy.NewImagepullpolicy(testTimeoutDuration, testPodNamespace, testPodName, testContainerCount)
-	assert.NotNil(t, handler)
+func TestLogging_ReelEof(t *testing.T) {
+	_, handlers, jsonParseResult, err := createTest()
+
+	assert.Nil(t, err)
+	assert.True(t, jsonParseResult.Valid())
+	assert.NotNil(t, handlers)
+
+	assert.Equal(t, 1, len(handlers))
+	handler := handlers[0]
+	// just ensure there isn't a panic
 	handler.ReelEOF()
-	// Todo: Write test.
 }
 
-// Test_Imagepullpolicy_ReelTimeout is the unit test for Imagepullpolicy}_ReelTimeout().
-func TestImagepullpolicy_ReelTimeout(t *testing.T) {
-	// Todo: Write test.
-}
+func TestLogging_ReelTimeout(t *testing.T) {
+	_, handlers, jsonParseResult, err := createTest()
+
+	assert.Nil(t, err)
+	assert.True(t, jsonParseResult.Valid())
+	assert.NotNil(t, handlers)
 
-// Test_Imagepullpolicy_ReelMatch is the unit test for Imagepullpolicy_ReelMatch().
-func TestImagepullpolicy_ReelMatch(t *testing.T) {
-	// Todo: Write test.
+	assert.Equal(t, 1, len(handlers))
+	handler := handlers[0]
+	assert.Nil(t, handler.ReelTimeout())
 }
 
-const (
-	testTimeoutDuration = time.Second * 1
-	testContainerCount  = 1
-	testInputError      = "Always"
-	testPodNamespace    = "testPodNamespace"
-	testPodName         = "testPodName"
-)
+func TestLogging_ReelMatch(t *testing.T) {
+	tester, handlers, jsonParseResult, err := createTest()
 
-var (
-	testInputSuccess = "IfNotPresent"
-)
+	assert.Nil(t, err)
+	assert.True(t, jsonParseResult.Valid())
+	assert.NotNil(t, handlers)
+
+	assert.Equal(t, 1, len(handlers))
+	handler := handlers[0]
+	step := handler.ReelMatch(expectedPassPattern, "", testInputSuccess)
```

Comment:
does this pass? i think you are using your input with the pattern from the logging test?